cmake/linker/xt-ld/linker_flags.cmake

```diff
@@ -1,3 +1,36 @@
+# Copyright (c) 2024 Nordic Semiconductor
+#
 # SPDX-License-Identifier: Apache-2.0
 
+check_set_linker_property(TARGET linker PROPERTY base
+                          ${LINKERFLAGPREFIX},--gc-sections
+                          ${LINKERFLAGPREFIX},--build-id=none
+)
+
+if(NOT CONFIG_MINIMAL_LIBCPP AND NOT CONFIG_NATIVE_LIBRARY AND NOT CONFIG_EXTERNAL_MODULE_LIBCPP)
+  set_property(TARGET linker PROPERTY cpp_base -lstdc++)
+endif()
+
+check_set_linker_property(TARGET linker PROPERTY baremetal
+    -nostdlib
+    -static
+    ${LINKERFLAGPREFIX},-X
+    ${LINKERFLAGPREFIX},-N
+)
+
+check_set_linker_property(TARGET linker PROPERTY orphan_warning
+			  ${LINKERFLAGPREFIX},--orphan-handling=warn
+)
+
+check_set_linker_property(TARGET linker PROPERTY orphan_error
+			  ${LINKERFLAGPREFIX},--orphan-handling=error
+)
+
 set_property(TARGET linker PROPERTY partial_linking "-r")
+
+check_set_linker_property(TARGET linker PROPERTY no_relax ${LINKERFLAGPREFIX},--no-relax)
+)
```

Comment:
Commit 9a9e252d1677c broke the build with every `xt-clang` toolchain we have. Any idea?

```
14:01:36 CMake Error at /srv/home/jenkins/workspace/sof_generic_build/zephyr/cmake/linker/xt-ld/linker_flags.cmake:32:
14:01:36   Parse error.  Expected a command name, got right paren with text ")".
14:01:36 Call Stack (most recent call first):
14:01:36   /srv/home/jenkins/workspace/sof_generic_build/zephyr/cmake/target_toolchain_flags.cmake:42 (include)
14:01:36   /srv/home/jenkins/workspace/sof_generic_build/zephyr/cmake/modules/kernel.cmake:148 (include)
14:01:36   /srv/home/jenkins/workspace/sof_generic_build/zephyr/cmake/modules/zephyr_default.cmake:142 (include)
14:01:36   /srv/home/jenkins/workspace/sof_generic_build/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
14:01:36   /srv/home/jenkins/workspace/sof_generic_build/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:97 (include_boilerplate)
14:01:36   CMakeLists.txt:5 (find_package)
```

Reply:
Nevermind, it's trivial. Fix incoming.